### DISCUSSION TITLE
How can I filter tests with stencil cli

### DISCUSSION THREAD

**Discussion by deleonio** [stenciljs/core#5225] [Stencil Testing]
Is it possible to filter tests, like this: `npx stencil test --spec --filter link.spec.ts`?

**Comment by christian-bromann** [comment]
Hey @deleonio 👋 

there is currently no such filter flag. Unfortunately I can't give you any workaround at this point. We started to revisit Stencils testing capabilities this quarter and actively working on improving the experience of testing Stencil components. I hope to be able to give you updates soon.

**Comment by rwaskiewicz** [comment]
I do believe you can achieve filtering, however, it's a jest argument rather than a Stencil argument. That means that you have to separate Stencil and Jest based arguments with a '--'.

For example: `npx stencil test --spec -- path/to/link.spec.ts`

**Reply by christian-bromann** [reply]
Interesting! I think this is worthwhile to document. I will raise a PR to the docs.

**Reply by christian-bromann** [reply]
https://github.com/ionic-team/stencil-site/pull/1316

**Reply by dgonzalezr** [reply]
I think you can skip the path to the test file and provide only the filename.

E.g: ‘npx stencil test --spec -- link.spec.ts‘

Same applies to e2e tests. We have used that way on our project without issues.
